src/screens/Catalog/CatalogScreen.tsx

```diff
@@ -1,21 +1,64 @@
-import React from 'react';
-import {SafeAreaView, ScrollView, StyleSheet} from 'react-native';
+import React, {useState} from 'react';
+import {
+  ActivityIndicator,
+  FlatList,
+  SafeAreaView,
+  StyleSheet,
+} from 'react-native';
 import {ScreenProps} from '../../navigation/TabNavigator.tsx';
 import SearchInput from '../../components/SearchInput.tsx';
-import {findBooks} from '../../api/Book.ts';
-import BookList from '../../components/BookList/BookList.tsx';
+import {getBooks} from '../../api/Book.ts';
+import {Book, Collection, CollectionMeta} from '../../types/types.ts';
+import BookListItem from '../../components/BookList/BookListItem.tsx';
+import {useInfiniteQuery} from '@tanstack/react-query';
 
 type CatalogProps = ScreenProps<'Catalog'>;
 
 function CatalogScreen({}: CatalogProps): React.JSX.Element {
-  const books = findBooks();
+  const [refreshKey, setRefreshKey] = useState(0);
+  const queryBooks = useInfiniteQuery<Collection<Book>, Error>({
+    queryKey: ['books', refreshKey],
+    queryFn: ({pageParam}) => {
```

Comment:
спробуй типи додати прямо в колбек, типу такого

```suggestion
    queryFn: ({pageParam}: CollectionMeta): void => {
```

Reply:
думаю, просто забув додати `void` тому в не вийшло, ще можне спробувати винести в useCallback саму функцію

Reply:
Додавати тип прямо в колбек вже пробував.
**Було:**
<img width="806" alt="image" src="https://github.com/ivanartamonov/voicebook/assets/7717669/18ab9109-ba3a-4a4e-a7e9-e62361f3d18a">
**Стало:**
<img width="830" alt="image" src="https://github.com/ivanartamonov/voicebook/assets/7717669/b9433e43-77d5-4fc4-a5fa-aad517976482">
В середені колбека тепер все гаразд, але саме поле `queryFn` тепер підксреслюється. При наведенні, там ось така пояснялка від IDE:
<img width="1063" alt="image" src="https://github.com/ivanartamonov/voicebook/assets/7717669/ccf80268-06b8-49ee-aadd-936b02bfd53f">
Наскільки я розумію, цей unknown зашитий десь прям в недрах React Query.